node/subsystem/src/messages.rs

```diff
@@ -255,6 +261,18 @@ pub enum DisputeParticipationMessage {
 	},
 }
 
+/// Messages going to the dispute distribution subsystem.
+#[derive(Debug)]
+pub enum DisputeDistributionMessage {
+
+	/// Tell dispute distribution to distribute an explicit dispute statement to
+	/// validators.
+	SendDispute(DisputeMessage),
+
+	/// Get receiver for receiving incoming network requests for dispute sending.
+	DisputeSendingReceiver(mpsc::Receiver<sc_network::config::IncomingRequest>),
```

Comment:
I wonder if this separate `mpsc::Receiver` can introduce any type of deadlock. It might be why we see `statement-distribution`, which is the other subsystem using this pattern, as the one of the only other subsystems which becomes unresponsive.

Reply:
I would not see how this might happen. They are both queried in their own tasks, not blocking the main subsystem at all.